### PR TITLE
chore: update supported node version for spfx

### PR DESCRIPTION
**Pull request by HuihuiWu-Microsoft**
ADO: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/15263893?src=WorkItemMention&src-action=artifact_link

### CODE REVIEW

File: packages/fx-core/templates/plugins/resource/spfx/solution/README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ The SharePoint Framework (SPFx) is a page and web part model that provides full
 
 ## Prerequisites
 
-> - Node.js v12/14/16
+> - Node.js v14/16
```
Comment (a1exwang) [inline packages/fx-core/templates/plugins/resource/spfx/solution/README.md:18]: Don't forget to update the sample code
https://github.com/OfficeDev/TeamsFx-Samples/blob/e9c78bfc1ebe61213837782caf4276f7c5ee8c36/todo-list-SPFx/README.md?plain=1#L21

File: packages/fx-core/src/common/deps-checker/internal/nodeChecker.ts [hunk 2]
```diff
@@ -179,7 +179,7 @@ export class SPFxNodeChecker extends NodeChecker {
   }
 
   protected async getSupportedVersions(): Promise<string[]> {
-    return ["12", "14", "16"];
+    return ["14", "16"];
```
Comment (a1exwang) [inline packages/fx-core/src/common/deps-checker/internal/nodeChecker.ts:182]: Also need to update env checker document?

https://github.com/OfficeDev/TeamsFx/blob/26d397a3713fab1d37364065874756ddea0399ca/docs/vscode-extension/envchecker-help.md?plain=1#L9
https://github.com/OfficeDev/TeamsFx/blob/26d397a3713fab1d37364065874756ddea0399ca/docs/vscode-extension/envchecker-help.md?plain=1#L60
https://github.com/OfficeDev/TeamsFx/blob/26d397a3713fab1d37364065874756ddea0399ca/docs/vscode-extension/envchecker-help.md?plain=1#L122

Comment (HuihuiWu-Microsoft) [reply packages/fx-core/src/common/deps-checker/internal/nodeChecker.ts:182]: Updated

File: packages/fx-core/templates/plugins/resource/spfx/solution/README.md (hunk 1 above)
Comment (HuihuiWu-Microsoft) [reply packages/fx-core/templates/plugins/resource/spfx/solution/README.md:18]: PR raised: https://github.com/OfficeDev/TeamsFx-Samples/pull/459. Thanks!

**Review by a1exwang — APPROVED** [review]

**Review by xu-hong — DISMISSED** [review]

**Review by a1exwang — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Pull request merged**